### PR TITLE
Update developer.rst

### PR DESCRIPTION
**Pull request by shreyneil**
This PR is regarding the following  https://github.com/weecology/retriever/issues/1016  issue.
Please verify and get back

### CODE REVIEW

**Comment by henrykironde** [comment]
Where did you get this explanation? Any reference

**Comment by shreyneil** [comment]
I referred to the explanation in the conda forge repo for retriever feedstock.

**Comment by henrykironde** [comment]
Could you post a link here

**Comment by henrykironde** [comment]
Thanks

**Comment by henrykironde** [comment]
I think may be considered as plagiarism,  How about you provide one or two sentences and link to the where you got the notes. 

I have found the same here

https://github.com/conda-forge/retriever-feedstock

**Comment by shreyneil** [comment]
I will make changes to the sentences sir , i got the notes from https://github.com/conda-forge/retriever-feedstock here.

File: docs/developer.rst [hunk 1]
```diff
@@ -144,6 +144,22 @@ Pull requests submitted to the repository will automatically be tested using
...
+
+Once the submission has been made, the changes will be executed across appropriate platforms to give the reviewer a chance to confirm that the changes made should result in a successful build.
+
+After merging of your PR, the recipe will be re-constructed and uploaded autonomously to the conda-forge channel, whereupon the built packages will be available for everyone to install and use from the channel. 
+
+All branches in the conda-forge/retriever-feedstock are built and the created packages are uploaded dynamically, so PRs should be based on branches in forks and branches in the main repository shall be used to build distinct package versions only.
+
+For producing a uniquely identifiable distribution:
+
+ If the version of a package is not being increamented, then the build/number can be added or increased .
```
Comment (zhangcandrew) [inline docs/developer.rst]: change increamented -> incremented

File: docs/developer.rst [hunk 2]
```diff
@@ -144,6 +144,22 @@ Pull requests submitted to the repository will automatically be tested using
...
+ If the version of a package is not being increamented, then remember to return the build/number back to 0.
```
Comment (zhangcandrew) [inline docs/developer.rst]: change increamented -> incremented

Comment (zhangcandrew) [reply docs/developer.rst]: I think it should be 

"If the version of a package is being incremented"

File: docs/developer.rst [hunk 3]
```diff
@@ -144,6 +144,22 @@ Pull requests submitted to the repository will automatically be tested using
 these systems and results reported in the ``checks`` section of the pull request
 page.
 
+Updating conda-forge retriever-feedstock
+========================================
+
+For improving the retriever recipe as well as for building a new package version, it is required to fork this repository https://github.com/conda-forge/retriever-feedstock and submit a PR.
```
Comment (zhangcandrew) [inline docs/developer.rst]: Change: 
```
it is required to fork this repository https://github.com/conda-forge/retriever-feedstock and submit a PR.
```
to
```
please fork `this repository <https://github.com/conda-forge/retriever-feedstock>`_ and submit a PR. 
```

File: docs/developer.rst [hunk 4]
```diff
@@ -144,6 +144,22 @@ Pull requests submitted to the repository will automatically be tested using
 these systems and results reported in the ``checks`` section of the pull request
 page.
 
+Updating conda-forge retriever-feedstock
+========================================
+
+For improving the retriever recipe as well as for building a new package version, it is required to fork this repository https://github.com/conda-forge/retriever-feedstock and submit a PR. 
+
+Once the submission has been made, the changes will be executed across appropriate platforms to give the reviewer a chance to confirm that the changes made should result in a successful build.
```
Comment (zhangcandrew) [inline docs/developer.rst]: Change: 
```
Once the submission has been made, the changes will be executed across appropriate platforms to give the reviewer a chance to confirm that the changes made should result in a successful build.
```
to
```
Once submitted, the changes will be executed across the appropriate platforms to ensure a successful build. 
```

File: docs/developer.rst [hunk 5]
```diff
@@ -144,6 +144,22 @@ Pull requests submitted to the repository will automatically be tested using
...
+
+After merging of your PR, the recipe will be re-constructed and uploaded autonomously to the conda-forge channel, whereupon the built packages will be available for everyone to install and use from the channel.
```
Comment (zhangcandrew) [inline docs/developer.rst]: Change:
```
After merging of your PR, the recipe will be re-constructed and uploaded autonomously to the conda-forge channel, whereupon the built packages will be available for everyone to install and use from the channel.
```
to
```
After merging your PR, the recipe will be re-constructed and uploaded autonomously to the conda-forge channel. The built packages should now be available for everyone to install and use from the channel.
```

File: docs/developer.rst [hunk 6]
```diff
@@ -144,6 +144,22 @@ Pull requests submitted to the repository will automatically be tested using
...
+Updating conda-forge retriever-feedstock
+========================================
+
+For improving the retriever recipe as well as for building a new package version, it is required to fork this repository https://github.com/conda-forge/retriever-feedstock and submit a PR. 
+
+Once the submission has been made, the changes will be executed across appropriate platforms to give the reviewer a chance to confirm that the changes made should result in a successful build.
+
+After merging of your PR, the recipe will be re-constructed and uploaded autonomously to the conda-forge channel, whereupon the built packages will be available for everyone to install and use from the channel. 
+
+All branches in the conda-forge/retriever-feedstock are built and the created packages are uploaded dynamically, so PRs should be based on branches in forks and branches in the main repository shall be used to build distinct package versions only.
```
Comment (zhangcandrew) [inline docs/developer.rst]: Change:
```
All branches in the conda-forge/retriever-feedstock are built and the created packages are uploaded dynamically, so PRs should be based on branches in forks and branches in the main repository shall be used to build distinct package versions only.
```
to
```
All branches in the conda-forge/retriever-feedstock are created and uploaded dynamically, so PRs should be based on branches in forks. Branches in the main repository shall be used to build distinct package versions only.
```

File: docs/developer.rst [hunk 7]
```diff
@@ -144,6 +144,22 @@ Pull requests submitted to the repository will automatically be tested using
...
+Once submitted, the changes will be executed across the appropriate platforms to ensure a successful build.
+
+After merging your PR, the recipe will be re-constructed and uploaded autonomously to the conda-forge channel. The built packages should now be available for everyone to install and use from the channel.
+
+All branches in the conda-forge/retriever-feedstock are created and uploaded dynamically, so PRs should be based on branches in forks. Branches in the main repository shall be used to build distinct package versions only.
+
+For producing a uniquely identifiable distribution:
+
+ If the version of a package is not being incremented, then the build/number can be added or increased .
+ If the version of a package is not being incremented, then remember to return the build/number back to 0.
```
Comment (zhangcandrew) [inline docs/developer.rst]: This needs to be changed from: 
```
If the version of a package is not being incremented, then remember to return the build/number back to 0.
```
to:
```
If the version of a package is being incremented, then remember to return the build/number back to 0
```

File: docs/developer.rst [hunk 8]
```diff
@@ -144,6 +144,22 @@ Pull requests submitted to the repository will automatically be tested using
 these systems and results reported in the ``checks`` section of the pull request
 page.
 
+Updating conda-forge retriever-feedstock
+========================================
+
```
Comment (zhangcandrew) [inline docs/developer.rst]: Add quotes around this entire paragraph

File: docs/developer.rst (hunk 7 above)
Comment (zhangcandrew) [inline docs/developer.rst]: Quotes end after line 161.

File: docs/developer.rst (hunk 8 above)
Comment (zhangcandrew) [inline docs/developer.rst]: Just to be safe, we should probably preface this paragraph that we took from conda-forge/retriever-feedstock by saying something like: 
```
To update the confa-forge retriever-feedstock, please reference the updating retriever-feedstock 
section from the 'conda-forge/retriever-feedstock <https://github.com/conda-forge/retriever-feedstock>'_ repository: 
"
```
and start the paragraph with quotes.

**Comment by zhangcandrew** [comment]
Hey @shreyneil, after discussing a bit with @henrykironde, we think it might be best to let people know where the info came from. So as a compromise we've decided including the link to the original instructions in the doc and then quoting the paragraph we wrote might be best. Let me know what you think.

**Comment by shreyneil** [comment]
@zhangcandrew  sounds good , i will make the necessary changes , just let me know where to include the link and how to frame the whole thing, Thanks.

**Comment by zhangcandrew** [comment]
Thanks @shreyneil! 

One thing I wanted to discuss with @henrykironde is that PR #1036 seems to already cover parts of this information in its own section. Moreover, it is right underneath where you want to append this section. I think the information is a bit redundant, since the merge we did for #1036 seems to already cover parts of what we want to put in this one. 

I think it'd be better if we condensed some of the information in this PR and moved it down into the "**Creating or Updating a Conda Release**" section. What do you think?

**Comment by shreyneil** [comment]
Sounds logical, just let me know about the Changes you want.

**Comment by zhangcandrew** [comment]
Sweet! 

Ok, let's go ahead and put the changes made in this PR into the section introduced in #1036. Basically just look over the section "**Creating or Updating a Conda Release**" and add into that anything that it doesn't include from this PR. 

Let me know if you have any questions!

**Comment by shreyneil** [comment]
@zhangcandrew  I think only these are some of the changes that are needed.
Please verify.

**Comment by zhangcandrew** [comment]
Looks good @shreyneil. 

Looking it over a final time, I think the intention of #1016 was to make sure developers new how to update or create packages for our retriever-feedstock specifically, so i've gone ahead and made some minor changes to the paragraph once more. (feel free to correct me if I'm wrong @henrykironde )

I think ultimately the paragraph should end up looking like this: 

```
To create or update a Conda Release, first fork the conda-forge `retriever-feedstock repository <https://github.com/conda-forge/retriever-feedstock>`_.
    
Once forked, open a pull request to the retriever-feedstock repository. Your package will be tested on Windows, Mac and Linux.
    
When your pull request is merged, the package will be rebuilt and become automatically available on conda-forge.

All branches in the conda-forge/retriever-feedstock are created and uploaded immediately, so PRs should be based on branches in forks. Branches in the main repository shall be used to build distinct package versions only.

For producing a uniquely identifiable distribution:

 - If the version of a package is not being incremented, then the build/number can be added or increased .
 - If the version of a package is being incremented, then remember to return the build/number back to 0.

```
This was tested locally and made sure to be formatted correctly. 

If you put that in, I think we'll be good!

**Comment by shreyneil** [comment]
Done with all the changes. Please verify.

**Comment by zhangcandrew** [comment]
Looks good to me, if anything else pops up from anyone, feel free to open this.

**Pull request merged**